File: Algorithms/Quick Sort/C/quick_sort.c
#include <stdio.h>
#include <stdbool.h>

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

int partition(int arr[], int low, int high) {
    int pivot = arr[low];
    int i = low + 1;
    int j = high;
    while (true) {
        while (i <= high && arr[i] <= pivot) i++;
        while (arr[j] > pivot) j--;
        if (i >= j) break;
        swap(&arr[i], &arr[j]);
    }
    swap(&arr[low], &arr[j]);
    return j;
}

void quick_Sort(int arr[], int low, int high) {
    if (low < high) {
        int pi = partition(arr, low, high);
        quick_Sort(arr, low, pi - 1);
        quick_Sort(arr, pi + 1, high);
    }
}

int main() {
    int arr[] = {10, 7, 8, 9, 1, 5};
    int size = sizeof(arr) / sizeof(arr[0]);
    quick_Sort(arr, 0, size - 1);
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    return 0;
}

File: Algorithms/Quick Sort/quick_sort.js
function partition(arr, low = 0, high = arr.length) {
  const pivot = arr[low];
  let i = low + 1;
  let j = high;
  while (i <= j) {
    while (arr[i] <= pivot) ++i;
    while (arr[j] > pivot) --j;
    if (i < j) {
      [arr[i], arr[j]] = [arr[j], arr[i]];
      ++i;
      --j;
    }
  }
  [arr[low], arr[j]] = [arr[j], arr[low]];
  return i;
}

function quick_Sort(arr, low = 0, high = arr.length - 1) {
  if (low < high) {
    const pi = partition(arr, low, high);
    quick_Sort(arr, low, pi - 1);
    quick_Sort(arr, pi + 1, high);
  }
  return arr;
}